azure/services/aso/aso.go

```diff
@@ -202,8 +204,15 @@ func (r *reconciler[T]) CreateOrUpdateResource(ctx context.Context, spec azure.A
 	}
 	parameters.SetAnnotations(annotations)
 
-	diff := cmp.Diff(existing, parameters)
-	if diff == "" {
+	log.V(2).Info("creating or updating resource", "diff", cmp.Diff(existing, parameters))
```

Comment:
The only user-visible part of this change should be this slightly lower log fidelity where we don't know for sure here which of create and update is being done here. That same info can still be derived from earlier log statements in this function without having to add any extra control flow.

Reply:
> To be clear we still have the ability to pass along that user-convenience (create vs update) but we are making an opinionated change (in the interest of simplifying the implementation) of relying upon preexisting flow control log statements to infer create vs update.
> 
> Is that right?

That's correct.

> I wonder if for the tags and labels stuff above we can drop the boolean `resourceExists` and just look for nil `existing` instead?

The bool exists because `existing` technically isn't a pointer, it's only a type that implements an interface, so the compiler won't let us compare it to `nil`.